lib/internal/main/check_syntax.js

```diff
@@ -45,16 +45,16 @@ if (process.argv[1] && process.argv[1] !== '-') {
   });
 }
 
-function checkSyntax(source, filename) {
+async function checkSyntax(source, filename) {
   const { getOptionValue } = require('internal/options');
   let isModule = false;
   if (filename === '[stdin]' || filename === '[eval]') {
     isModule = getOptionValue('--input-type') === 'module';
   } else {
     const { defaultResolve } = require('internal/modules/esm/resolve');
     const { defaultGetFormat } = require('internal/modules/esm/get_format');
-    const { url } = defaultResolve(pathToFileURL(filename).toString());
-    const { format } = defaultGetFormat(url);
+    const { url } = await defaultResolve(pathToFileURL(filename).toString());
```

Comment:
Can it improve IO if we also switch the local module to async IO?

Reply:
i'm not sure what improve IO means exactly in this context, but the refactoring of doing so is significant and shouldn't be done in this PR. various parts of the resolvers seem shared with ESM and CJS now with the package import/exports bits and I faced trouble with even having ESM resolve/getFormat becoming async due to CJS dependencies.